packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,191 @@
+import { LoanMasterNodeRegTestContainer } from './loan_container'
+import { GenesisKeys } from '@defichain/testcontainers'
+import BigNumber from 'bignumber.js'
+import { TestingGroup } from '@defichain/jellyfish-testing'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('Loan', () => {
+  const tGroup = TestingGroup.create(2, i => new LoanMasterNodeRegTestContainer(GenesisKeys[i]))
+  let vId = ''
+  let collateralAddr = ''
+
+  beforeAll(async () => {
+    await tGroup.start()
+    await tGroup.get(0).container.waitForWalletCoinbaseMaturity()
+
+    const { vaultId, collateralAddress } = await setup()
+    vId = vaultId
+    collateralAddr = collateralAddress
+  })
+
+  async function setup (): Promise<any> {
+    // token setup
+    const collateralAddress = await tGroup.get(0).container.getNewAddress()
+    await tGroup.get(0).token.dfi({ address: collateralAddress, amount: 20000 })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+    await tGroup.get(0).token.create({ symbol: 'BTC', collateralAddress })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+    await tGroup.get(0).token.mint({ symbol: 'BTC', amount: 20000 })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // oracle setup
+    const addr = await tGroup.get(0).generateAddress()
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'BTC', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' }
+    ]
+    const oracleId = await tGroup.get(0).rpc.oracle.appointOracle(addr, priceFeeds, { weightage: 1 })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '10000@BTC', currency: 'USD' }] })
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '2@TSLA', currency: 'USD' }] })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // collateral token
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      priceFeedId: oracleId
+      // activateAfterBlock: 130  // <- hit socket hang up
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    await tGroup.get(0).rpc.loan.setCollateralToken({
+      token: 'BTC',
+      factor: new BigNumber(1),
+      priceFeedId: oracleId
+    })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+
+    // loan token
+    await tGroup.get(0).rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      priceFeedId: oracleId
+    })
+    await tGroup.get(0).generate(1)
```

Comment:
I am thinking after generating block(s), we will write `await tGroup.waitForSync()`. Maybe there is a need to add a function in `TestingGroup` so programmer just needs to write 1 line of code rather than writing 2 lines.

Reply:
https://github.com/DeFiCh/jellyfish/pull/576#discussion_r686481633